### ISSUE TITLE
New Rule: require-exhaustive-switch

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#41]
## Research Evidence

From empirical LLM bug studies (20.77% misinterpretations, 15.27% missing corner cases): LLMs miss handling enum variants or union type members in switch statements.

## Description

For TypeScript union types and enums, require switch statements to cover all cases. Similar to TypeScript's `noSwitchExhaustiveChecks` but as an ESLint rule with LLM-friendly error messages.

### Bad
```ts
type Status = 'pending' | 'active' | 'inactive';

function getLabel(status: Status) {
  switch (status) {
    case 'pending': return 'Pending';
    case 'active': return 'Active';
    // Missing 'inactive'!
  }
}
```

### Good
```ts
function getLabel(status: Status) {
  switch (status) {
    case 'pending': return 'Pending';
    case 'active': return 'Active';
    case 'inactive': return 'Inactive';
  }
}
```

### With Explicit Unhandled
```ts
function getLabel(status: Status) {
  switch (status) {
    case 'pending': return 'Pending';
    case 'active': return 'Active';
    case 'inactive': 
      throw new Error('Not implemented yet');
  }
}
```

## Why This Matters for LLMs
- Catches when LLMs forget enum variants
- Works with TypeScript's type system
- Forces explicit handling of all cases

## Priority
High - catches a major LLM bug pattern
